### ISSUE TITLE
Create seperate setting for terminal background & terminal background ansi color

### ISSUE THREAD

**Issue by AlbertMarashi** [zed-industries/zed#17313]
### Check for existing issues

- [X] Completed

### Describe the feature

When using a `transparent` or `blurred` theme, I wish to have a terminal that is transparent.

Unfortunately, text using the "background" color in my terminal gets turned invisible.

![image](https://github.com/user-attachments/assets/7ea57953-e00a-4732-8b06-3dbab9522d24)

The correct behavior should look like this

![image](https://github.com/user-attachments/assets/6ba28b9c-7ccc-43ac-b123-cab4ee5d15b8)

However, our terminal is turned dark and we can no longer see the blurred background effect in the theme

**Comment by AlbertMarashi** [comment]
I have been able to debug the issue, it appears that vitest is using a specific ansi color code

```
echo -e "\033[7m\033[1m\033[36m RUN \033[39m\033[22m\033[27m"
```

This command will output:

- `\033[7m`: Enables reverse video (swaps foreground and background colors)
- `\033[1m`: Enables bold text
- `\033[36m`: Sets the text color to cyan
- RUN: The actual text being displayed
- `\033[39m`: Resets the text color to default
- `\033[22m`: Turns off bold
- `\033[27m`: Turns off reverse video

I think it has something to do with the reverse video - it's putting `terminal.background` (`#0000`) as the `terminal.foreground` equivalent - so when it prints, its appearing as invisible text inside of Zed's terminal

Is this a bug?

**Comment by AlbertMarashi** [comment]
Additionally, pasted text into the terminal appears as white, presumably for the same "video reversal" reason 

![image](https://github.com/user-attachments/assets/78bce5c7-148b-430c-9656-8907c22343c1)

What is the right approach here?

**Comment by AlbertMarashi** [comment]
```bash
#!/bin/bash

# Function to print colored text with reverse video
print_reverse_color() {
    local color=$1
    local color_name=$2
    echo -e "\033[7m\033[1m\033[${color}m ${color_name} \033[39m\033[22m\033[27m"
}

# Print header
echo "Reverse Video ANSI Color Test"
echo "-----------------------------"

# Standard colors
print_reverse_color "30" "Black"
print_reverse_color "31" "Red"
print_reverse_color "32" "Green"
print_reverse_color "33" "Yellow"
print_reverse_color "34" "Blue"
print_reverse_color "35" "Magenta"
print_reverse_color "36" "Cyan"
print_reverse_color "37" "White"

echo

# Bright colors
print_reverse_color "90" "Bright Black"
print_reverse_color "91" "Bright Red"
print_reverse_color "92" "Bright Green"
print_reverse_color "93" "Bright Yellow"
print_reverse_color "94" "Bright Blue"
print_reverse_color "95" "Bright Magenta"
print_reverse_color "96" "Bright Cyan"
print_reverse_color "97" "Bright White"

echo

# Default color
echo -e "\033[7m\033[1m Default Color \033[22m\033[27m"
```

With `terminal.background` set to `#0000`


![image](https://github.com/user-attachments/assets/c2de4ed6-5a92-4a88-8246-56e9ad300a40)


With `terminal.background` set to `#000`

![image](https://github.com/user-attachments/assets/a694a7e1-9fc6-4cfd-a197-bb11e37d12d0)
